llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -2137,10 +2137,9 @@ OpenMPIRBuilder::InsertPointTy OpenMPIRBuilder::createReductions(
   unsigned NumReductions = ReductionInfos.size();
   Type *RedArrayTy = ArrayType::get(Builder.getPtrTy(), NumReductions);
   Builder.SetInsertPoint(AllocaIP.getBlock()->getTerminator());
-  //  Builder.restoreIP(AllocaIP);
   Value *RedArray = Builder.CreateAlloca(RedArrayTy, nullptr, "red.array");
 
-  Builder.SetInsertPoint(InsertBlock, InsertBlock->end());
+  Builder.SetInsertPoint(InsertBlock, InsertBlock->begin());
```

Comment:
Why did you change this one to go at the beginning when it used to be at the end? Couldn't you set the insertion point just before the terminator instead?

Reply:
I must have confused myself - I thought I'd changed that earlier, but that must have been another place...